src/translate.py

```diff
@@ -1354,6 +1355,71 @@ def format(self, fmt: Formatter) -> str:
         return self.get_var_name()
 
 
+@dataclass
+class SwitchControl:
+    control_expr: Expression
+    jump_table: Optional[GlobalSymbol] = None
+    offset: int = 0
+
+    def format(self, fmt: Formatter) -> str:
+        index = format_expr(self.control_expr, fmt)
+        if not self.jump_table:
+            return index
+        if self.offset:
+            index = f"{index} - {self.offset}"
+        return f"(&{format_expr(self.jump_table, fmt)})[{index}]"
```

Comment:
might be misleading to include the offset in the output here, since it's already been added to each case

I think it's fair to give up on compilability in the --gotos-only case and print something like `switch (expr); // switch N`; then we can remove this format method altogether. (maybe with the jump table in a comment if you think it's useful)

Reply:
Yeah, I'll give up compatibility with `--gotos-only` then. :+1: